Automattic/vip-go-mu-plugins
# adds cron job to detect inactive index-versions

## Description


## Changelog Description

### Plugin Updated: Search

We added inactive index detection. This will detect all the inactive index versions if there was no change in active/non-active index versions for more than a month. This is the first step towards having an automated cleanup of these non-used index versions.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test
I couldn't find a nice way to test the cron task so what I end up doing is:

1. Switch from cron to running the code always by adding `do_action( self::CRON_EVENT_NAME );` in `VersioningCleanupJob::init`.
2. Then you will probably fake some data. You can use this to setup one inactive version`lando wp eval "update_option( 'vip_search_index_versions', [ 'post' => [ 1 => ['number' => 1, 'created_time' => 1610662421 ], 2 => ['number' => 2, 'created_time' => 1610662421, 'activated_time' => 1610662421, 'active' => true ] ]] );"`
3. Finally you will probably not see the logs, so adding something like `var_dump($message);` in `VersioningCleanupJob::send_notification` will make you see what's up.
